File: app/src/main/java/com/example/petridelivery/lists/viewmodels/ClientViewModel.java
package com.example.petridelivery.lists.viewmodels;

import android.view.View;

import com.example.petridelivery.app.PetriDeliveryApp;
import com.example.petridelivery.lists.datasources.ClientPagingSource;
import com.example.petridelivery.lists.viewmodels.abs.BaseViewModel;
import com.example.petridelivery.wrappers.ClientWrapper;
import com.petri.delivery.web.objects.ClientDto;

import lombok.Getter;

@Getter
public class ClientViewModel extends BaseViewModel<ClientDto>{
    public static final String PAGE_SIZE_PROP = "client.pagination.pageSize";

    public ClientViewModel(ClientWrapper clientWrapper, PetriDeliveryApp app, View view) {
        super(new ClientPagingSource(clientWrapper, view), app, PAGE_SIZE_PROP);
    }
}

File: app/src/main/java/com/example/petridelivery/lists/viewmodels/ClientViewModel.kt
package com.example.petridelivery.lists.viewmodels

import com.example.petridelivery.app.PetriDeliveryApp
import com.example.petridelivery.lists.datasources.ClientPagingSource
import com.example.petridelivery.lists.viewmodels.abs.IntegerKeyViewModel
import com.example.petridelivery.wrappers.ClientWrapper
import com.petri.delivery.web.objects.ClientDto


class ClientViewModel(clientWrapper: ClientWrapper, app: PetriDeliveryApp) : IntegerKeyViewModel<ClientDto>(ClientPagingSource(clientWrapper), app, PAGE_SIZE_PROP) {
    companion object {
        val PAGE_SIZE_PROP: String = "client.pagination.pageSize"
    }
}